tooling/nargo_fmt/tests/input/impl.nr

```diff
@@ -14,8 +14,6 @@ impl MyType {
     fn method(self: Self) {}
 
     fn method(mut self: Self) {}
-
-    fn method(&mut self: Self) {}
```

Comment:
why was this removed?

Reply:
Ah, good catch! I don't know how you find these things in the giant diff, but it's great 😄 It seems that never parsed, I can't remember if the old parser had the same issue. This meant that the file was not formatted, and the test passed because input and expected were the same. I captured a personal note to work on this separately, but I'll work on it now because I think it should be easy to fix.

Reply:
I just tried it with the chumsky parser (I checked out one commit before the handwritten parser was merged) and this program:

```noir
struct Foo;

impl Foo {
    fn foo(&mut self: Self) {

    }
}
```

gives this error:

```noir
error: Unexpected Colon, expected one of ), ,
  ┌─ src/main.nr:4:21
  │
4 │     fn foo(&mut self: Self) {
  │                     -
```

so it never worked.

Do you think we should fix it in this PR or in a separate one? I don't know what's the reason that it doesn't work, but I don't know if you do something like `fn foo(&mut self: SomeOtherType)`... should that compile?

Reply:
I do this mistake in Rust all the time: shouldn't it be `fn foo(self: &mut Self)`?

Reply:
Interesting. It seems the old parser (and I copied it for the new parser) parses `&mut self` into a pattern named `self` with type `&mut Self`. So I guess it's a shortcut to write that? And I just tried writing `fn(&mut self: Self)` in Rust and it also gives an error saying it's expecting `,` instead of `:`, so maybe there's nothing to do here.

I'm also starting to remember that I might have added that `&mut self: Self` line in that input file while I was coding the new parser, to make sure it parsed well... but I didn't realize that it was an error, and those errors don't show up in the formatter tests, oops! 😊